### DISCUSSION TITLE
Data binding doesn't get changes

### DISCUSSION THREAD

**Discussion by n2-holden** [unoplatform/uno#10780] [Q&A]
x:Bind get the initial value of the property I'm binding to, but the value in the control it doesn't show a change when the property changes.  This is probably a dumb question as I'm new to Uno.

**Comment by jeromelaban** [comment]
You can find the documentation for x:Bind [here](https://platform.uno/docs/articles/features/windows-ui-xaml-xbind.html?q=bind), with links to the official Microsoft documentation.

It's likely that you need to set the binding mode to `OneWay`, as the default mode for `x:Bind` is `OneTime`. This is different from `{Binding}` which is `OneWay` by default.

**Reply by n2-holden** [reply]
I must have something wrong in the project.  Using Text="{x:Bind TestTitleLabel, Mode=OneWay}" gives me an error in the editor that says: The property Mode was not found in type 'Bind'

**Reply by jeromelaban** [reply]
See this article for the resolution: https://platform.uno/docs/articles/get-started-wizard.html#the-xaml-editor-shows-the-type-page-does-not-support-direct-content-message

**Reply by n2-holden** [reply]
The problem isn't that the editor shows an error when it shouldn't.  Setting the Mode-OneWay doesn't get the Text to update in the TextBlock object.

**Reply by jeromelaban** [reply]
I see. Can you create a repro project that shows what you're trying to do and attach it here?

**Reply by n2-holden** [reply]
I found the problem.  I needed to add a DependencyProperty.  Why is that some examples show a simple a property like:

```csharp
public string TestTitleLabel { set; get; }
```

and other times they show:

```csharp
public static readonly DependencyProperty TestTitleLabelProperty = DependencyProperty.Register("TestTitleLabel", typeof(string), typeof(AppState), new PropertyMetadata(null));
public string TestTitleLabel
{
         get { return (string)GetValue(TestTitleLabelProperty); }
         set { SetValue(TestTitleLabelProperty, value); }
}
```

I got used to the simple form in Blazor, but I'm vary familiar with the DependencyProperty pattern from WPF

**Reply by jeromelaban** [reply]
Indeed, if you want to use simple properties, you'll need to use a `INotifyPropertyChanged` implementation. A simple and good one recently is [MVVM toolkit](https://learn.microsoft.com/en-us/windows/communitytoolkit/mvvm/observableobject). 

You can otherwise use the `Bindings.Update()` generated method to force a refresh of the x:Bind bindings when a property value changes, but it's generally best to avoid it as it can refresh too much.

**Reply by n2-holden** [reply]
Thanks.  It's all starting to make sense.
